### PR TITLE
Use AVA for tests

### PR DESCRIPTION
**Pull request by li-kai**
### AVA
Simple enough, just removed mocha and chai and changed the tests to run with AVA.

### Problems getting flow working
No problems with getting flow annotations, they'll get stripped out automatically
Thing is, `.flowconfig` is the only way to specify where the flow parsing server runs, and there aren't any runtime options or flags to do so. So, there is no way to stop flow from checking files in `/test/` when running `npm start`, so you end up with a bunch of import errors (as webpack doesn't resolve /test/).

Similarly, since there's no way to stop flow from checking files in `src`, errors would pop up there too. I think flow still needs some more time maturing haha. Having multiple `.flowconfig` at different places isn't a really good solution.

### CODE REVIEW

File: v3/.eslintrc.js [hunk 1]
```diff
@@ -14,12 +14,6 @@ module.exports = {
     "import",
     "flowtype"
   ],
-  "parserOptions": {
```
Comment (yangshun) [inline v3/.eslintrc.js]: how come these are not needed anymore?

File: v3/test/actions/helpers.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import test from 'ava';
+import * as actions from 'actions/helpers';
+
+function testShouldReturnArr(t, type, action) {
+  const payload = [];
+  const expectedResult = {
+    type,
+    payload: payload,
```
Comment (yangshun) [inline v3/test/actions/helpers.js]: can use es6 object literal shorthand.

File: v3/test/actions/helpers.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+test('resetRequestState should convert item to an array', (t) => {
+  testShouldReturnArr(t, actions.RESET_REQUEST_STATE, actions.resetRequestState);
+});
+
+test('resetErrorState should convert item to an array', (t) => {
+  testShouldReturnArr(t, actions.RESET_ERROR_STATE, actions.resetErrorState);
+});
+
+
```
Comment (yangshun) [inline v3/test/actions/helpers.js]: Extra new line here. Hmm seems like the linting doesn't happen in the tests?

If the goal is to categorize the tests, you can leave a comment instead.

File: v3/test/actions/moduleBank.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import test from 'ava';
+import * as actions from 'actions/moduleBank';
+
+test('fetchModuleList should dispatch a request', (t) => {
+  t.true(typeof actions.fetchModuleList() === "function");
```
Comment (yangshun) [inline v3/test/actions/moduleBank.js]: singlequotes for strings.

File: v3/test/example.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import test from 'ava';
+
+test('should always pass', (t) => {
+    t.pass();
```
Comment (yangshun) [inline v3/test/example.js]: 2 spaces

File: v3/.eslintrc.js (hunk 1 above)
Comment (li-kai) [inline v3/.eslintrc.js]: [SurviveJS says to remove them](http://survivejs.com/webpack/advanced-techniques/linting/#connecting-eslint-with-babel). I also found no problems running the build so I reckoned it is fine to do so.

**Pull request merged**